frontend/src/FriendSelect.tsx

```diff
@@ -53,26 +53,33 @@ const FriendSelect = (props: IProps): JSX.Element => {
     currentValue.value = c.value;
   }
 
-  preloadedCards.forEach((c) => {
-    if (
+  const notTrumpFilter: (c: ICardInfo) => boolean = (c: ICardInfo) => {
+    return (
       c.number !== null &&
       c.number !== rank &&
       (props.trump.Standard == null || c.typ !== props.trump.Standard.suit)
-    ) {
-      // exclude highest card
-      if (
-        (props.friend_selection_policy === "HighestCardNotAllowed" &&
-          ((rank !== "A" && c.number !== "A") ||
-            (rank === "A" && c.number !== "K"))) ||
-        props.friend_selection_policy === "Unrestricted"
-      ) {
-        cardOptions.push({
-          label: `${c.number}${c.typ}`,
-          value: c.value,
-        });
-      }
-    }
-  });
+    );
+  };
+  const policyFilters: { [s: string]: (c: ICardInfo) => boolean } = {
+    PointCardNotAllowed: (c: ICardInfo) => c.points === 0,
+    HighestCardNotAllowed: (c: ICardInfo) => {
+      return (
+        (rank !== "A" && c.number !== "A") || (rank === "A" && c.number !== "K")
+      );
+    },
+    Unrestricted: (_c: ICardInfo) => true,
+  };
+  const policyFilter: (c: ICardInfo) => boolean =
```

Comment:
Pretty cool that the Typescript linter can figure out that this will work!

Reply:
Yeah, this lint error really surprised me! 😄